cpp/tests/community/ecg_test.cu

```diff
@@ -31,42 +30,105 @@ TEST(ecg, success)
       1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0,
       1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0};
 
-  gdf_column col_off, col_ind, col_w;
+  int num_verts = off_h.size() - 1;
+  int num_edges = ind_h.size();
+
+  std::vector<int> cluster_id (num_verts, -1);
+
+  rmm::device_vector<int>    offsets_v(off_h);
+  rmm::device_vector<int>    indices_v(ind_h);
+  rmm::device_vector<float>  weights_v(w_h);
+  rmm::device_vector<int>    result_v(cluster_id);
+
+  cugraph::experimental::GraphCSR<int,int,float> graph_csr(offsets_v.data().get(),
+                                                           indices_v.data().get(),
+                                                           weights_v.data().get(),
+                                                           num_verts,
+                                                           num_edges);
+
+  ASSERT_NO_THROW((cugraph::nvgraph::ecg<int32_t, int32_t, float>(graph_csr, .05, 16, result_v.data().get())));
+
+  cudaMemcpy ((void*) &(cluster_id[0]), result_v.data().get(), sizeof(int)*num_verts, cudaMemcpyDeviceToHost);
```

Comment:
cluster_id[0] = result_v[0]; should work. thrust::device_vector and thrust::device_ptr (note that thrust device_ptr is different from a raw pointer) properly handles data transfer between host and device.

Reply:
That didn't work, but I changed it to std::copy in the latest commit.

Reply:
What do you mean by this didn't work.

thrust::device_vector<int> d_ints(100, 0.0);
thrust::host_vector<int> h_ints = d_ints;

Something like the above is what I have used extensively. Could you show the code that didn't work?

Reply:
Ah... so I need to change cluster_id from being std::vector to being thrust::host_vector.  Missed that part of your comment.